### ISSUE TITLE
What can impact the speed of tests ran with Jest?

### ISSUE THREAD

**Issue by thisissami** [kentcdodds/ama#233]
I got `Jest` working in my environment, but I'm fairly certain that it's running way slower than it's supposed to. 

Currently, I'm just running 2 tests that are copied from the Jest intro tutorials. One is [a simple sum function][https://facebook.github.io/jest/docs/getting-started.html#content], and another is [a simple checkbox example using `Enzyme`][https://facebook.github.io/jest/docs/tutorial-react.html#dom-testing]. They take on average about 10 seconds until they start running (showing `Determining test suites to run...` for that time), and then about another 10 seconds from that point onwards (once the timer starts). This seems absurdly long for such simple components... 

I tried setting my `testEnvironment` to `node` to speed things up, but it hasn't made much of a difference. 

So accordingly: 

1) Is this normal? 

2) If not - what can cause `Jest` to slow down like this? 

**Some potential factors?**

1) I _am_ running these tests within a larger `React` project that contains ~20 or so react components, but none of them are being referenced by any of the tests - so I don't think they should be impacting this. =/

2) I'm using the `fileName.test.js` naming system to let `Jest` know where the test files are, instead of putting everything in a `__tests__` folder as per the tutorials.

**Comment by kentcdodds** [comment]
Hi there @thisissami,
I'm sorry this has taken me so long to respond to! Here are a few thoughts (@cpojer may have more thoughts as well).

1. If it's taking `10 seconds` that's weird. There's probably something wrong...
2. Jest has a lot of really great features which make it really attractive as a testing framework. With all of those features come some overhead. You're likely going to see Jest taking longer to start up and run your tests than you would a simple node script using `assert`
3. Jest is _really_ fast, but you normally don't see dividends until you have ~100 tests or so (that's a made up number). Once you get more and more tests, things like the intelligent watch mode and running only tests impacted by your changes become even more awesome.
4. Try installing [watchman](https://facebook.github.io/watchman/) to get better watch mode speed.

I hope these things are helpful. Just remember that Jest is really fast and if it's going slow, it's probably something wrong with your tests and if you fix it, your tests should run faster. I've found this to be true for my own stuff. Good luck!

**Comment by cpojer** [comment]
Jest doesn't really have any overhead while running tests, so from the moment you see "RUNS" until that test finishes, that time is entirely spent in your test. Everything before or after may be Jest. As @kentcdodds pointed out, Jest has some startup overhead but we'll look at improving this. With the changes to the watch mode in Jest 19, it will be instant and you won't have any re-run overhead (ok, well, it's 30ms max until your test starts running ;) ).

**Comment by kentcdodds** [comment]
Looking forward to Jest 19! So much awesome always coming from you and other Jest contributors!

**Comment by kentcdodds** [comment]
One thing I failed to mention is it can be a good idea to use `--runInBand` on CI. My tests went from timing out after 1.5 hours to about 4 minutes with `--runInBand`! YMMV. [Read More](https://facebook.github.io/jest/docs/troubleshooting.html#tests-are-extremely-slow-on-docker-and-or-continuous-integration-server)

**Comment by thisissami** [comment]
Ah sorry for the incredibly belated noticing of this response!! I'm not sure how this completely slipped me. 

Well since I first asked this question, the contract I was working on where I first began using `Jest` wrapped up. I've since began (and wrapped up) a second `React` contract, and I pushed heavily to have solid unit-testing before building anything, specifically using `Jest`. They were super down with that, and I can say with certainty that my second run-around with `Jest` was very speedy and without any issues. :) 

My original contract involved getting a giant repository to work with full of dependencies of all kinds and a bulky webpack configuration. _(the newer contract was a very lean & simple repository)_ Additionally, the first company I worked for required me to use `React v0.13`, which could have had something to do with the slowdown? _(The second company wanted me to use the latest version)_

Thanks for taking the time to respond to this @kentcdodds & thanks for adding your $0.02 @cpojer. The watch mode in `Jest v19` has been really incredible & a pleasure to use. :)
